### ISSUE TITLE
ComboBox autocomplete does not work

### ISSUE THREAD

**Issue by daniellittledev** [microsoft/fluentui#19379]
### Environment Information

- @fluentui/react 8.26.2

### Please provide a reproduction of the bug in a codepen:

I'm using the combo-box and trying get autocomplete behaviour to work. See the code sandbox for the stark difference between the MaterialUI ComboBox which works flawlessly and the fluent ui version. The textbox part of the ComboBox also looks broken without allowFreeform as you can start to type but when a selection is made backspace cannot be used and effectively locks in the selection blocking autocomplete.

https://codesandbox.io/s/combo-box-comparison-ldmlw

User Demo:
![AutoComplete](https://user-images.githubusercontent.com/1761789/129158250-9f87aeed-01b6-4fef-9617-310dabd4241b.gif)

#### Actual behavior:

The ComboBox only opens when the user starts typing, and does not filter the options as they type.

#### Expected behavior:

The ComboBox should open as soon as the control is focused and the options should be filtered as the user starts to enter text.

### Priorities and help requested:

Are you willing to submit a PR to fix? I do not know enough to answer.

Requested priority: Normal? It does seem to indicate autocomplete does not work.

Products/sites affected: Product In development.

**Comment by gouttierre** [comment]
@daniellittledev - Thanks for filing this issue with us and showing the willingness to help out. To set expectation our team will have a look at the issue but only once capacity allows - but we will loop you in on the fix, if found to be necessary. 
  
@bsunderhus -  Would you be able to confirm if this is a regression, or if this behavior is an issue?

**Comment by imatiach-msft** [comment]
@gouttierre We also see this behavior in our app, it's really bad
https://github.com/microsoft/responsible-ai-toolbox/issues/767
Is there any way to improve the auto complete?

**Comment by imatiach-msft** [comment]
also pinging @GeoffCoxMSFT

**Comment by imatiach-msft** [comment]
are there any workarounds to this issue?  Should we be using a different component for the combo box?  Our users are complaining about this behavior.

**Comment by trdand** [comment]
Similar to this issue, when freeform is set to false, I am not able to use the textbox at all
Even when trying to type value in the options.

**Comment by NWH-SAmin5** [comment]
@gouttierre @GeoffCoxMSFT any updates on this?

**Comment by JustSlone** [comment]
Updating v8 for fuzzy search is out of scope right now, we are refreshing and fixing a lot of issues with [combobox in v9](https://react.fluentui.dev/?path=/docs/preview-components-combobox--default) 

@microsoft/cxe-coastal can you all look into #3 and #4 above. Let's fix the know bugs and glitches with combobox autocomplete if we can without breaking folks.

**Comment by smhigley** [comment]
For anyone following this issue, PR #25602 adds an `allowFreeInput` prop to Combobox that allows the user to type while the combobox is in focus, and selects a matching value or resets to the last selected value when blurred.

I'd love any feedback on whether that would address the issues raised in this PR, if anyone has time to take a look. You can preview the behavior here, in the "ComboBox with allowFreeInput" example: https://fluentuipr.z22.web.core.windows.net/pull/25602/public-docsite/index.html#/controls/web/combobox
